File: cli/banner.py
"""Startup banner for Brix."""

from __future__ import annotations

from rich.console import Console
from rich.table import Table

BRIX_ASCII = r"""
 ██████╗ ██████╗ ██╗██╗  ██╗
 ██╔══██╗██╔══██╗██║╚██╗██╔╝
 ██████╔╝██████╔╝██║ ╚███╔╝
 ██╔══██╗██╔══██╗██║ ██╔██╗
 ██████╔╝██║  ██║██║██╔╝ ██╗
 ╚═════╝ ╚═╝  ╚═╝╚═╝╚═╝  ╚═╝
"""


def show_banner(console: Console, model: str, version: str, cwd: str) -> None:
    """Print the startup banner with session info using Rich Console."""
    console.print(BRIX_ASCII, style="bold cyan")
    console.print("  BRIX — Personal AI Agent\n", style="dim")

    table = Table(show_header=False, box=None, padding=(0, 2))
    table.add_column(style="dim")
    table.add_column(style="white")
    table.add_row("Model", model)
    table.add_row("Version", version)
    table.add_row("Directory", cwd)
    console.print(table)
    console.print()
    console.print("  Type /help for commands · Ctrl+C to exit\n", style="dim")

File: src/cli/banner.ts
import chalk from 'chalk'

/**
 * BRIX 启动横幅
 * 显示 ASCII 艺术字、模型、版本、工作目录等信息
 */

const BRIX_ASCII = `
 ██████╗ ██████╗ ██╗██╗  ██╗
 ██╔══██╗██╔══██╗██║╚██╗██╔╝
 ██████╔╝██████╔╝██║ ╚███╔╝
 ██╔══██╗██╔══██║██║ ██╔██╗
 ██████╔╝██║  ██║██║██╔╝ ██╗
 ╚═════╝ ╚═╝  ╚═╝╚═╝╚═╝  ╚═╝
`

/**
 * 显示启动横幅
 * @param model - 当前使用的模型名称
 * @param version - Brix 版本号
 * @param cwd - 当前工作目录
 */
export function showBanner(model: string, version: string, cwd: string): void {
  console.log(chalk.bold.cyan(BRIX_ASCII))
  console.log(chalk.dim('  BRIX — Personal AI Agent\n'))
  console.log(chalk.dim('  Model:    ') + chalk.white(model))
  console.log(chalk.dim('  Version:  ') + chalk.white(version))
  console.log(chalk.dim('  Directory: ') + chalk.white(cwd))
  console.log()
  console.log(chalk.dim('  Type /help for commands · Ctrl+C to exit\n'))
}
